### PR TITLE
Adding go versions 1.18 and 1.19 dev images.

### PR DESCRIPTION
**Pull request by jmacelroy**
The golang:1 image will now be based off of 1.19 instead of 1.17.
Also, images golang:1.18 and golang:1.19 will be generated with
their respective versions.

### CODE REVIEW

**Review by AgustinRamiroDiaz — APPROVED** [review]

**Review by rberrelleza — APPROVED** [review]

**Comment by maroshii** [comment]
I think in 1.18 the behaviour of go get changes:

> In earlier versions of Go, 'go get' was used to build and install packages.
Now, 'go get' is dedicated to adjusting dependencies in go.mod. 'go install'
may be used to build and install commands instead. When a version is specified,
'go install' runs in module-aware mode and ignores the go.mod file in the
current directory

File: golang/Dockerfile [hunk 1]
```diff
@@ -1,14 +1,14 @@
-ARG VERSION=1.19
+ARG VERSION=1.16
```
Comment (rberrelleza) [inline golang/Dockerfile:1]: I think we should deprecate 1.16 and stick to 1.17-1.19 at the most

**Review by maroshii — APPROVED** [review]

File: golang/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-ARG VERSION=1.18
+ARG VERSION=1.16
```
Comment (rberrelleza) [inline golang/Dockerfile:1]:
```suggestion
ARG VERSION=1.19
```

File: docker-compose.golang.yml [hunk 3]
```diff
@@ -7,8 +7,15 @@ services:
       context: .
       dockerfile: golang/Dockerfile
       args:
-        VERSION: 1.18
-
+        VERSION: 1.19
```
Comment (pchico83) [inline docker-compose.golang.yml:10]: 1.18? 1.19 is defined bellow

**Review by pchico83 — APPROVED** [review]

File: docker-compose.golang.yml (hunk 3 above)
Comment (AgustinRamiroDiaz) [reply docker-compose.golang.yml:10]: @pchico83 This is for the `okteto/golang:1` tag which should always point to the latest `1.x`

Comment (pchico83) [reply docker-compose.golang.yml:10]: ah, perfect

**Review by pchico83 — APPROVED** [review]

**Comment by AgustinRamiroDiaz** [comment]
I think @jmacelroy won't be checking this. Should we claim ownership of this?

**Pull request merged**